tests/utils/test_shell_util.py

```diff
@@ -142,5 +139,43 @@ def test_it_should_log_unexpected_errors_as_errors(self):
         self.assertEquals(mock_logger.warn.call_count, 0)
 
 
+class RunCommandTestCase(AgentTestCase):
+    def test_run_command_it_should_run_without_errors(self):
+        command = ["echo", "42"]
+
+        with patch("azurelinuxagent.common.utils.shellutil.logger", autospec=True) as mock_logger:
+            ret = shellutil.run_command(command)
+            self.assertEquals(ret, "42\n")
+            self.assertEquals(mock_logger.error.call_count, 0)
+
+    def test_run_command_it_should_log_and_raise_an_exception_from_command(self):
+        command = ["ls", "nonexistent_file"]
+        expected_returncode = 2
+
+        with patch("azurelinuxagent.common.utils.shellutil.logger", autospec=True) as mock_logger:
+            with self.assertRaises(Exception):
+                shellutil.run_command(command)
+
+            self.assertEquals(mock_logger.error.call_count, 1)
+
+            logged_error_message = u"Command: [{0}], return code: [{1}]".format(command, expected_returncode)
+            self.assertIn(logged_error_message, mock_logger.error.call_args_list[0][0][0])
+
+    def test_run_command_it_should_log_and_raise_an_exception_from_subprocess_popen(self):
+        command = ["ls", "nonexistent_file"]
+
+        with patch("azurelinuxagent.common.utils.shellutil.logger", autospec=True) as mock_logger:
+            with patch("azurelinuxagent.common.utils.shellutil.subprocess.Popen",
```

Comment:
suggestion: instead of mocking Popen I'd just pass it a command that does not exist

Reply:
Thanks, that's seems like a cleaner way to do it.